### PR TITLE
Updated cache docs

### LINKED ISSUE #124: Docs on caching are out of date
#### Do you want to request a *feature* or report a *bug*?

This is a documentation bug.

#### What is the current behavior?

Visiting the [Padrino caching documentation](http://padrinorb.com/guides/features/padrino-cache/) I am told to specify the cache as follows:

```
Padrino.cache = Padrino::Cache::Store::Memory.new(50)
```

Following those instructions yields the following error:

```
NameError: uninitialized constant Padrino::Cache::Store
```

The above applies to the creation of all different types of store.

#### What is the expected behavior?

The documentation should tell me to create a cache following the new approach:

```
Padrino.cache = Padrino::Cache.new(:LRUHash)
```

The documentation in the [padrino-cache module](https://github.com/padrino/padrino-framework/tree/master/padrino-cache) worked for me.

#### Which versions of Ruby, Padrino, Sinatra, Rack, OS are you using? Did this work in previous versions?

- Ruby 2.3.1
- Padrino 0.13.3.2

I've not tested it for other versions, as this is a new development for us.

### PR DESCRIPTION
**Pull request by spikeheap**
This PR pulls in https://github.com/padrino/padrino-framework/blob/master/padrino-cache/README.rdoc to update the caching documentation and resolve #124.

### CODE REVIEW

File: guides/03_features/04_padrino-cache.html.md [hunk 1]
```diff
@@ -7,412 +7,354 @@ title: Padrino Cache
...
-  register Padrino::Cache
-  enable :caching
-
-  get '/post/:id', :cache => true do
-    @post = Post.find(params[:id])
-    cache_key :my_name
-  end
-end
-```
+    # Basic, page-level caching
```
Comment (wikimatze) [inline guides/03_features/04_padrino-cache.html.md]: I would leave the word "Basic"

File: guides/03_features/04_padrino-cache.html.md [hunk 2]
```diff
@@ -7,412 +7,354 @@ title: Padrino Cache
...
 
 You can set a global caching option or a per app caching options.
 
 ### Global Caching Options
 
-```ruby
-Padrino.cache = Padrino::Cache::Store::Memcache.new(::Memcached.new('127.0.0.1:11211', :exception_retry_limit => 1))
-Padrino.cache = Padrino::Cache::Store::Memcache.new(::Dalli::Client.new('127.0.0.1:11211', :exception_retry_limit => 1))
-Padrino.cache = Padrino::Cache::Store::Redis.new(::Redis.new(:host => '127.0.0.1', :port => 6379, :db => 0))
-Padrino.cache = Padrino::Cache::Store::Memory.new(50)
```
Comment (wikimatze) [inline guides/03_features/04_padrino-cache.html.md]: Why is `Memory`  gone from the examples?

Comment (spikeheap) [inline guides/03_features/04_padrino-cache.html.md]: Hi @wikimatze, `Memory` is now `LRUHash`. That's not particularly intuitive, so I've added a comment by each `LRUHash` reference saying it's in-memory.

**Comment by spikeheap** [comment]
Hi @wikimatze, I've addressed your comments, and removed a copyright notice I copied in erroneously. What do you think?

File: guides/03_features/04_padrino-cache.html.md [hunk 3]
```diff
@@ -7,412 +7,350 @@ title: Padrino Cache
...
 
-You can manage your cache from anywhere in your app:
-
-```ruby
-MyApp.cache.set('val', 'test')
-MyApp.cache.get('val') # => 'test'
-MyApp.cache.delete('val')
-MyApp.cache.flush
-```
+#### Application Caching Options
```
Comment (adam12) [inline guides/03_features/04_padrino-cache.html.md]: Probably an artifact from the migration but would you mind dropping this down a header  level? (`###`).

**Comment by adam12** [comment]
Looks good to me. Thanks for the contribution @spikeheap!

**Comment by spikeheap** [comment]
Hi @adam12, thanks for taking a look. I've addressed that heading, so it should be good to merge now. I'll back-port these changes to padrino-framework so it's a straight copy/paste for future changes.

**Comment by wikimatze** [comment]
Everything is fine so, except that you have taken `===` instead of `###` for the headlines. All other guides have the `#` for headings format. Could you change this?

**Comment by spikeheap** [comment]
Hi @wikimatze, sorry I don't follow – I've not had quite enough coffee yet, but I'm pretty sure this PR has `###`-style headings. Could you link me to a line and I'll take a look? Were you looking at https://github.com/padrino/padrino-framework/issues/2107, which uses the `===` syntax?

**Comment by wikimatze** [comment]
Everything is fine, the rdoc files had the syntax, the rest is fine.

**Pull request merged**